src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -912,11 +913,13 @@ protected function asDateTime($value)
         // Finally, we will just assume this date is in the format used by default on
         // the database connection and use that format to create the Carbon object
         // that is returned back out to the developers after we convert it here.
-        if (Date::hasFormat($value, $format)) {
-            return Date::createFromFormat($format, $value);
+        try {
+            $date = Date::createFromFormat($format, $value);
+        } catch (InvalidArgumentException $_) {
```

Comment:
```suggestion
        } catch (InvalidArgumentException $e) {
```

Reply:
`_` is a common code for "unused". Else I would rather use "exception" instead of a 1-letter variable. But I do not care too much.